lasagne/layers/dnn.py

```diff
@@ -10,15 +8,27 @@
 from .normalization import BatchNormLayer
 from ..utils import as_tuple
 
-if not theano.sandbox.cuda.cuda_enabled:
-    raise ImportError(
-            "requires GPU support -- see http://lasagne.readthedocs.org/en/"
-            "latest/user/installation.html#gpu-support")  # pragma: no cover
-elif not dnn.dnn_available():
+from lasagne import theano_backend
+dnn_avail = False
+if theano_backend == 'pygpu':
+    from theano.gpuarray import dnn
+    if dnn.dnn_present():
+        dnn_avail = True
+elif theano_backend == 'pygpu_sandbox':
+    from theano.sandbox.gpuarray import dnn
+    if dnn.dnn_present():
+        dnn_avail = True
+elif theano_backend == 'cuda_sandbox':
+    from theano.sandbox.cuda import dnn
+    if dnn.dnn_available():
+        dnn_available = True
+else:
+    raise ImportError("requires GPU support -- see http://lasagne.\
+                readthedocs.org/en/latest/user/installation.html#gpu-support")
+if not dnn_available:
     raise ImportError(
-            "cuDNN not available: %s\nSee http://lasagne.readthedocs.org/en/"
-            "latest/user/installation.html#cudnn" %
-            dnn.dnn_available.msg)  # pragma: no cover
+        "cuDNN not available: %s\nSee http://lasagne.readthedocs.org\
+        /en/latest/user/installation.html#cudnn")  # pragma: no cover
```

Comment:
I wasn't sure that there is such message for ```dnn_present()```. I will double check and try to keep the message.